pkg/truncindex/truncindex.go

```diff
@@ -24,6 +24,46 @@ var (
 	ErrIllegalChar = errors.New("illegal character: ' '")
 )
 
+// ErrIDAlreadyExists is returned if the id is already in the TruncIndex
+type ErrIDAlreadyExists struct {
+	id string
+}
+
+func (e ErrIDAlreadyExists) Error() string {
+	return fmt.Sprintf("id already exists: %s", e.id)
+}
+
+// ErrInsertionFailed is returned if something went wrong inserting the given
+// id in the TruncIndex
+type ErrInsertionFailed struct {
+	id string
+}
+
+func (e ErrInsertionFailed) Error() string {
+	return fmt.Sprintf("failed to insert id: %s", e.id)
+}
+
+// ErrNoSuchID is returned if the TruncIndex doesn't contain the given id
+type ErrNoSuchID struct {
```

Comment:
I can't say that I like this. It's sorta different in stdlib, you're passed that ID, why do you need it in error(remeber that truncindex is "library" and you shouldn't rely on your knowledge of calling code now)? So, I think `var ErrNoSuchID = errors.New` would be enough.